rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictReturnExprRector.php

```diff
@@ -80,12 +80,26 @@ public function refactor(Node $node): ?Node
         return $node;
     }
 
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersion::PHP_74;
+    }
+
     private function hasSingleStrictReturn(ClassMethod $classMethod): bool
     {
         if ($classMethod->stmts === null) {
             return false;
         }
 
+        if ($this->betterNodeFinder->hasInstancesOf($classMethod->stmts, [Yield_::class])) {
+            return false;
+        }
+
+        $returns = $this->betterNodeFinder->findInstanceOf($classMethod->stmts, Return_::class);
```

Comment:
should exclude inner function, there is `findInstancesOfInFunctionLikeScoped` method for it:

```suggestion
        $returns = $this->betterNodeFinder->findInstancesOfInFunctionLikeScoped($classMethod, Return_::class);
```

Reply:
Thanks. Could you send those 2 changes in PR?

Reply:
I will try.

Reply:
https://github.com/rectorphp/rector-src/pull/2566